sync3/caches/user.go

```diff
@@ -210,10 +212,14 @@ func (c *UserCache) Unsubscribe(id int) {
 	delete(c.listeners, id)
 }
 
+// OnRegistered is called after the sync3.Dispatcher has successfully registered this
+// cache to receive updates. We use this to run some final initialisation logic that
+// is sensitive to race conditions; confusingly, most of the initialisation is driven
+// externally by sync3.SyncLiveHandler.userCache.
```

Comment:
I think this is accurate(?)

Reply:
Yes, `OnRegistered` and spaces handling came much later in development hence the weird multiple places. This _wasn't_ folded into one function because `OnRegistered` has taken out a global lock on the dispatcher so it's important that we don't spend too long inside this function.